### PR TITLE
change imagemagick lib to node-gm

### PR DESCRIPTION
**Pull request by frankrousseau**
It was the opportunity to perform a small refactoring by centralizing all calls to image magick in the same module.

### CODE REVIEW

File: server/controllers/file.coffee [hunk 1]
```diff
@@ -121,12 +94,12 @@ module.exports.createPhoto = (req, res, next) ->
 
         stream.on 'end', =>
             if not photo.binary.thumb?
-                resize rawFile, photo, 'thumb', (err) ->
+                thumbHelpers.resize rawFile, photo, 'thumb', (err) ->
                     return next err if err
-                    resize rawFile, photo, 'screen', (err) ->
+                    thumb.resize resize rawFile, photo, 'screen', (err) ->
```
Comment (aenario) [inline server/controllers/file.coffee]: something wrong here

File: server/controllers/file.coffee [hunk 2]
```diff
@@ -121,12 +94,12 @@ module.exports.createPhoto = (req, res, next) ->
...
                         fs.unlink rawFile, ->
                             res.send 201, photo
             else
-                resize rawFile, photo, 'screen', (err) ->
+                thumb.resize rawFile, photo, 'screen', (err) ->
```
Comment (aenario) [inline server/controllers/file.coffee]: should be thumbHelpers ?

File: server/helpers/thumb.coffee [hunk 3]
```diff
@@ -1,46 +1,97 @@
...
-                        console.log unlinkErr if err
+            attachFile = (err) =>
+                if err
+                    callback err
+                else
+
+            gmRunner = gm(srcPath).options(imageMagick: true)
+
+            if name is 'thumb'
+                gmRunner.crop(300, 300, 0, 0)
```
Comment (aenario) [inline server/helpers/thumb.coffee]: are you sure gm.crop actually do resize & crop ?

File: server/helpers/thumb.coffee [hunk 4]
```diff
@@ -1,46 +1,97 @@
...
+                        date: data.Properties['date:create']
+
+                callback null, metadata
+
+
+    # Attach given binary to given file. `name` is the binary field that will
+    # store file content.
+    attachFile: (file, dstPath, name, callback) ->
+        file.attachBinary dstPath, {name}, (err) ->
+            fs.unlink dstPath, (unlinkErr) ->
```
Comment (aenario) [inline server/helpers/thumb.coffee]: handle attachBinary err

**Comment by frankrousseau** [comment]
This time it should be fine.

**Comment by aenario** [comment]
Ok, everything looks good, lets wait for travis.

**Pull request merged**